File: 1160-letter-tile-possibilities/1160-letter-tile-possibilities.cpp
class Solution {
public:
int n;
    void solve(string& tiles,vector<bool>& res , unordered_set<string>& result, string& curr)
    {
        result.insert(curr);
        for(int i = 0 ; i< n ; i++)
        {
            if(res[i])
            continue;

            curr.push_back(tiles[i]);
            res[i]=true;
           
            solve(tiles,res,result,curr);

            res[i]=false;
            curr.pop_back();
        }
    }
    int numTilePossibilities(string tiles) {
     n = tiles.length();
     vector<bool> res (n,false);
     unordered_set<string> result;
     string curr="";
     solve(tiles,res,result,curr);
     return result.size()-1;   
    }
    
};

File: 1160-letter-tile-possibilities/1160-letter-tile-possibilities.js
/**
 * @param {string} tiles
 * @return {number}
 */
var numTilePossibilities = function(tiles) {
   let count = 0;
   const vec = new Array(26).fill(0);
   for(const ch of tiles)
   {
    vec[ch.charCodeAt(0)-'A'.charCodeAt(0)]++;

   }
   const solve = (vec)=>{
    count++;
    for(let i= 0;i<26;i++)
    {
        if(vec[i]===0)
        {
            continue;
        }
        vec[i]--;
        solve(vec);
        vec[i]++;
    }
   };
   solve(vec);
   return count -1; 
};